### PR TITLE
Update result factory to handle Algolia response

### PR DESCRIPTION
**Pull request by bwangYext**
Discussed with product, we are passing objectID for now since it is the only general field on Algolia response.

TEST=auto
npm run test

### CODE REVIEW

File: src/core/models/resultfactory.js [hunk 1]
```diff
@@ -45,6 +45,11 @@ export default class ResultFactory {
         continue;
       }
 
+      if (source === 'ALGOLIA') {
+        results.push(ResultFactory.fromAlgoliaSearchEngine(data));
+        continue;
+      }
+
```
Comment (Delerme) [inline src/core/models/resultfactory.js]: I think we've hit the point where we should probably refactor these 4 `if`s into a switch statement with a default case. Do you mind taking care of that in this commit?

Comment (bwangYext) [reply src/core/models/resultfactory.js]: Done.

File: src/core/models/resultfactory.js [hunk 2]
```diff
@@ -115,6 +115,21 @@ export default class ResultFactory {
     });
   }
 
+  /**
+   * Converts an API result object into a result view model.
+   * Maps view model fields based on the API data for a Algolia Search Engine object.
+   * Deatiails field is set to objectID since response has only one general field objectID.
```
Comment (Delerme) [inline src/core/models/resultfactory.js]: typo

**Review by Delerme — COMMENTED** [review]
nice thanks

File: src/core/models/resultfactory.js (hunk 2 above)
Comment (bwangYext) [reply src/core/models/resultfactory.js]: Done.

**Review by Delerme — APPROVED** [review]

**Review by b1lly — COMMENTED** [review]
Are there any tests we need to add to the ResultFactory?

**Comment by bwangYext** [comment]
> Are there any tests we need to add to the ResultFactory?

Currently there are only tests for truncate().

**Pull request merged**